### PR TITLE
Move `_compare_version` function, remove `version.py`, and clean up `__init__`

### LINKED ISSUE #3669: Move `_compare_version` function and remove `version.py`
it's a bit weird to have both `version.py` and `_version.py`, and `version.py` isn't really related to nilearn's version anymore. not necessarily in this PR, it would be good to move `_compare_version` somewhere in `_utils`, remove `version.py`, and define `__version__` directly in `__init__.py` (by importing `_version.py`)

_Originally posted by @jeromedockes in https://github.com/nilearn/nilearn/pull/3635#discussion_r1162771602_

### PR DESCRIPTION
**Pull request by ymzayek**
Closes #3669 and addresses https://github.com/nilearn/nilearn/pull/3635#discussion_r1158996543.

Changes proposed in this pull request:

- `version.py` is removed
- `_compare_version` function is moved to `_utils.helpers` (is there a more suitable location?)
- `__version__` is defined in `__init__.py`
- addresses https://github.com/nilearn/nilearn/pull/3635#discussion_r1158996543 
  - package was pulled according to https://github.com/ContinuumIO/anaconda-issues/issues/11294

### CODE REVIEW

**Review by Remi-Gau — APPROVED** [review]
LGTM.

File: doc/introduction.rst [hunk 1]
```diff
@@ -1,11 +1,3 @@
-.. for doc tests to run with recent NumPy 1.14, we need to set print options
```
Comment (jeromedockes) [inline doc/introduction.rst:1]: is this not the case anymore?

File: nilearn/__init__.py [hunk 2]
```diff
@@ -40,18 +40,13 @@
...
 import warnings
 
-from .version import __version__
-
-# TODO: Check if still needed
-# Workaround issue discovered in intel-openmp 2019.5:
-# https://github.com/ContinuumIO/anaconda-issues/issues/11294
-#
-# see also https://github.com/scikit-learn/scikit-learn/pull/15020
-os.environ.setdefault("KMP_INIT_AT_FORK", "FALSE")
```
[the remark this answers is not in the document]
Comment (ymzayek) [reply nilearn/__init__.py:54]: IIUC from this thread/comment it seems the package causing the issue was pulled: https://github.com/ContinuumIO/anaconda-issues/issues/11294#issuecomment-534134600

File: doc/introduction.rst (hunk 1 above)
Comment (ymzayek) [reply doc/introduction.rst:1]: Now we support numpy>=1.19. Plus there are no more doc tests in this file

**Review by jeromedockes — APPROVED** [review]

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by ymzayek** [comment]
I added a test to check the version but I'm not sure how to properly test nilearn/__init__.py#L48-L49 to resolve the codecov patch failure. Any suggestions?

**Comment by ymzayek** [comment]
Otherwise I'll merge by tomorrow

**Comment by jeromedockes** [comment]
> I added a test to check the version but I'm not sure how to properly test nilearn/__init__.py#L48-L49 to resolve the codecov patch failure. Any suggestions?

you could monkeypatch importing `nilearn._version` to make sure it fails:
```python
@pytest.fixture
def prevent_importing_version(monkeypatch):
    with monkeypatch.context():
        monkeypatch.delitem(sys.modules, "nilearn", False)
        _builtins_import = builtins.__import__

        def mock_import(name, *args, **kwargs):
            if name == "_version":
                raise ImportError()
            return _builtins_import(name, *args, **kwargs)

        monkeypatch.setattr("builtins.__import__", mock_import)
        yield
    import nilearn
    importlib.reload(nilearn)


def test_missing_version(prevent_importing_version):
    import nilearn
    assert nilearn.__version__ == "0+unknown"
```

but I'm not sure it's worth it

**Comment by ymzayek** [comment]
@jeromedockes ok if you agree it's not worth it then I will merge this but thanks for the tip! I need to get more familiar with monkey patching and mocking so that's quite helpful.

**Pull request merged**